src/AspNetStandard.Diagnostics.HealthChecks/Entities/HealthCheckResult.cs

```diff
@@ -23,13 +23,11 @@ public HealthCheckResult(HealthStatus status, string description = null, Excepti
 
         public override bool Equals(object obj)
         {
-            if (obj == null || !(obj is HealthCheckResult))
+            if (!(obj is HealthCheckResult item))

```

Comment:
```suggestion
            if (obj is not HealthCheckResult item)

```

Dependendo da versão do C#, dá para fazer isso ai.

Reply:
o HC tá na 7.3 :(

Reply:
acho que o **not** chegou só na 9